packages/sit-onyx/src/i18n/index.ts

```diff
@@ -0,0 +1,135 @@
+import { computed, inject, provide, unref, type InjectionKey, type MaybeRef } from "vue";
+import enUS from "./locales/en-US.json";
+
+/** Available translations that are used by Onyx components. */
+export type Translation = typeof enUS;
+
+export type ProvideI18nOptions = {
+  /**
+   * Current locale / language to use.
+   * If a ref is passed (e.g. the locale from the `vue-i18n` package)
+   * all Onyx messages will be updated if it changes (if locale is supported).
+   * If a message is missing for your currently set locale, English will be used as fallback.
+   *
+   * @default "en-US"
+   */
+  locale?: MaybeRef<string>;
+  /**
+   * Available translations / messages. English is always supported. For build-in translations, see:
+   * https://onyx.schwarz/i18n/
+   *
+   * @example
+   * ```ts
+   * import deDE from "sit-onyx/locales/de-DE.json";
+   * {
+   *   messages: {
+   *     // English is always supported so we don't need to add it here
+   *     'de-DE': deDE
+   *   }
+   * }
+   * ```
+   */
+  messages?: Record<string, DeepPartial<Translation> | Record<string, TranslationValue>>;
+};
+
+const I18N_INJECTION_KEY = Symbol() as InjectionKey<ReturnType<typeof createI18n>>;
+
+/**
+ * Creates a new i18n instance.
+ */
+const createI18n = (options?: ProvideI18nOptions) => {
+  /**
+   * Current locale.
+   * @default "en-US"
+   */
+  const locale = computed(() => unref(options?.locale) ?? "en-US");
+
+  const messages = computed(() => {
+    if (options?.messages && locale.value in options.messages) {
+      return options.messages[locale.value];
+    }
+    return enUS;
+  });
+
+  /**
+   * Gets the translation for the given key.
+   * If message is not found for current locale, English fallback will be used.
+   * @param placeholders Placeholders that will be replaced in the message string
+   * For pluralization, you must provide the placeholder `n`.
+   */
+  const t = computed(() => {
+    return (
+      key: ObjectToDottedStrings<Translation>,
+      placeholders: Record<string, string | number | undefined> = {},
+    ): string => {
+      // use English message as fallback
+      let message = resolveMessage(key, messages.value) ?? resolveMessage(key, enUS) ?? "";
+
+      if (placeholders) {
+        Object.entries(placeholders).forEach(([key, value]) => {
+          if (value === undefined) return;
+          // "gi" is used to replace all occurrences because String.replaceAll() is not available
+          // in our specified EcmaScript target
+          message = message.replace(new RegExp(`{${key}}`, "gi"), value.toString());
+        });
+      }
+
+      const pluralizationValue = typeof placeholders.n === "number" ? placeholders.n : undefined;
+      return resolvePluralization(message, pluralizationValue);
+    };
+  });
+
+  return { locale, t };
+};
+
+/**
+ * Provides a global i18n instance that is used by Onyx.
+ * Must only be called once in the `App.vue` file.
```

Comment:
```suggestion
 * Must only be called once in the `App.vue` file of a project that consumes Onyx.
```
maybe? onyx itself does not have one, might be misleading for contributors ¯\_(ツ)_/¯

Reply:
btw, how are we even able to inject i18n in onyx wihtout ever providing it 😵‍💫
looking forward to trying it out in a sample project ^^